libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/Ask.cs

```diff
@@ -35,7 +36,31 @@ public class Ask : SendActivity
 
         public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc, object options = null, CancellationToken cancellationToken = default)
         {
+            //get number of retries from memory
+            if (!dc.GetState().TryGetValue(DialogPath.Retries, out int retries))
+            {
+                retries = 0;
+            }
+
+            dc.GetState().TryGetValue(TurnPath.DIALOGEVENT, out DialogEvent trigger);
+            
+            if (dc.GetState().TryGetValue(DialogPath.ExpectedProperties, out List<string> lastExpectedProperties)
+                && ExpectedProperties.Where(prop => !lastExpectedProperties.Contains(prop)).ToList().Count == 0
+                && lastExpectedProperties.Where(prop => !ExpectedProperties.Contains(prop)).ToList().Count == 0)
+            {
+                if (trigger.Name.Equals("ask"))
```

Comment:
I don't think you need this test.  This is the Ask class and if someone derived from this and called it might be named differently.